File: benchmark/web/php/symfony-6.4-php-8.3/src/Controller/AppController.php
<?php

namespace App\Controller;

use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
use Symfony\Component\HttpFoundation\JsonResponse;
use Symfony\Component\Routing\Annotation\Route;
use Symfony\Component\HttpFoundation\Request;
use Symfony\Contracts\HttpClient\HttpClientInterface;

class AppController extends AbstractController
{
    private $client;

    public function __construct(HttpClientInterface $client)
    {
        $this->client = $client;
    }

    #[Route('/api/v1/periodic-table/element')]
    public function getElement(Request $request): JsonResponse
    {
        $symbol = $request->query->get('symbol');
        $response = $this->client->request('GET', 'http://web-data-source/element.json');
        $json_data = $response->toArray();
        $entry = $json_data[$symbol] ?? null;

        return $this->json([
            'name' => $entry['name'],
            'number' => $entry['number'],
            'group' => $entry['group'],
        ]);
    }

    #[Route('/api/v1/periodic-table/shells')]
    public function getShells(Request $request): JsonResponse
    {
        $symbol = $request->query->get('symbol');
        $response = $this->client->request('GET', 'http://web-data-source/shells.json');
        $json_data = $response->toArray();

        return $this->json([
            'shells' => $json_data[$symbol],
        ]);
    }
}

File: benchmark/web/dart/serinus-0.4-aot-3.3/lib/app_controller.dart
import 'package:serinus/serinus.dart';

import 'app_provider.dart';

class AppController extends Controller {
  AppController({super.path = '/api/v1/periodic-table'}) {
    on(Route.get('/element'), _handleElement);
    on(Route.get('/shells'), _handleShells);
  }

  Future<Response> _handleElement(RequestContext context) async {
    final String symbol = context.queryParameters['symbol'];
    final Map<String, dynamic> element = await context.use<AppProvider>().getElement(symbol);

    return Response.json(element);
  }

  Future<Response> _handleShells(RequestContext context) async {
    final String symbol = context.queryParameters['symbol'];
    final Map<String, dynamic> shells = await context.use<AppProvider>().getShells(symbol);
    return Response.json(shells);
  }
}
